packages/@glimmer/util/lib/untouchable-this.ts

```diff
@@ -0,0 +1,38 @@
+import { DEBUG } from '@glimmer/env';
+import { HAS_NATIVE_PROXY } from './platform-utils';
+
+export default function buildUntouchableThis(source: string): null | object {
+  let context: null | object = null;
+  if (DEBUG && HAS_NATIVE_PROXY) {
+    let assertOnProperty = (property: string | number | symbol) => {
+      throw new Error(
+        `You accessed \`this.${String(
+          property
+        )}\` from a function passed to the ${source}, but the function itself was not bound to a valid \`this\` context. Consider updating to usage of \`@action\`.`
```

Comment:
What is the right path for users in Glimmer.js here? I doubt `@action` is right, is it?

Reply:
ah, yes. We could say "a bound function", and give an arrow function as an example?